lib/guisso.ex

```diff
@@ -1,6 +1,21 @@
+# This behaviour only exists so we can mock Guisso during tests.
+defmodule Guisso.Client do
+  @callback enabled?() :: Bool.t()
+  @callback cookie_name :: String.t()
+  @callback sign_out(conn :: Plug.Conn.t(), redirect_url :: String.t()) :: Plug.Conn.t()
+  @callback sign_up(conn :: Plug.Conn.t(), redirect_url :: String.t()) :: Plug.Conn.t()
+  @callback request_auth_code(conn :: Plug.Conn.t(), redirect_url :: String.t()) :: Plug.Conn.t()
+  @callback authorize_url(conn :: Plug.Conn.t()) :: Plug.Conn.t()
+  @callback request_auth_token(conn :: Plug.Conn.t(), _ :: map()) :: {:ok, String.t(), String.t(), String.t()} | {:error, term()}
+  @callback sign_csrf_token(conn :: Plug.Conn.t(), client_id :: String.t(), expiration :: term()) :: String.t()
+  @callback verify_csrf_token(conn :: Plug.Conn.t(), state :: String.t()) :: :ok | {:error, String.t()}
+end
```

Comment:
Declaring an Elixir Behaviour for the client is the only way to mock the library (using Mox) in a test suite. There is no real value, except maybe for adding type declarations.

There is an usage example in Surveda's test suite, where Guisso can be enabled or not, hence it's useful to test both situations, as well as testing potential returns:

- The mock: https://github.com/instedd/surveda/pull/2039/files#diff-b47b0d37e5550ff01abb05fb968b9e1092ab551b4af063a075849857b59f3325
- Example usage: https://github.com/instedd/surveda/pull/2039/files#diff-3833c72b42d5fef028e477e596f7b407bb1ae3f5f0793018731a00748d813974